_posts/2017-06-02-jslang-meetup-async-await.markdown

```diff
@@ -0,0 +1,82 @@
+You want to do that too, in your city? Just start a meetup, take our katas that we solve and make it happen.
+
+## Our Scope and the Kata
+I announced the meetup about two weeks ago. There was no #jslang meetup since April 2016, more than one year ago. 
+But since I joined HolidayCheck in May 2017 and it is in my DNA to promote things I enjoy, I had to do restart this 
+meetup. The language JavaScript is moving forward and we need to keep up with what it offers.
+Async functions are one of the most interesting things that is changing in the language. In order to understand 
+them in the JS context I prepared a [short slide set sumarizing the state the async functions feature][slides]. 
+That was all of presentation we had. I set the context and introduced [the kata][kata] that we want to work on. 
+In the kata we want to combine two asynchronous functionalities, 1) getting the geolocation and 2) loading the 
+(nearest) airports depending on that result.
+The interesting part is actually not (only) the implementation but it already starts with the tests. Have you ever 
+had trouble writing asynchronous tests? Do this kata and practice even more!
+
+## How we did it
+We were twelve people at the event. And a quick hand-sign and we all agreed on setting up two mobs and solve the 
+problem by mob-programming. We had two mobs of 6 people each. Thanks to [Chris Neuroth][chris] who facilitated the 
+mob-programming for us. The most important rule I took away from it was that the one sitting at the keyboard should 
+not act unless told to.
+Both teams started and solve the first task, which was to implement the kata by using promises. This was the first 
+step in order to get familiar with the kata and also to feel the pain of using promises and test them properly. 
+Because every team also had the ambition to write good tests, and boy they did. This took about one hour. After that 
+we took a five minutes break and did some physical activity, playing table tennis. The table tennis challenge was 
+that we had to pass on one of the four rackets to the person who had none, while running around the table.
+Back to the kata, the next step was to move the tests to use async+await keywords, which mostly replaces promises 
+and makes tests more readable. But it has it's own challenges.
+In step three, which at this time kind of merged with step 2 we used async+await in all places. We learned about 
+async functions this way and naturally put `await` where it felt right. I had the impression this did not really 
+cause many problems and made it very easy to understand how async+await work. This was also the intention of this 
+meetup: allow us to learn without breaking a sweat and without falling asleep at some talk, but doing it.
+
+## The Discussion afterwards
+The last part after about 1,5 hours of mob-programming we used to discuss aspects of the kata. Topics we touched on were:
+- The source code of the two mobs
```

Comment:
Have you checked if this renders correctly? AFAIK this does not work in standard markdown, as titles, lists and code fences must be surrounded by blank lines. GFM however supports this.

See also the documentation of [this markdown-lint rule](https://github.com/markdownlint/markdownlint/blob/8da483ee194977ed109f1b12c7e72e0498ca084b/docs/RULES.md#md032---lists-should-be-surrounded-by-blank-lines):

> [...]  some parsers, including kramdown, will not parse lists that don't have blank lines before and after them